src/core/lib/transport/metadata_batch.cc

```diff
@@ -99,12 +98,11 @@ static grpc_error* maybe_link_callout(grpc_metadata_batch* batch,
 
 static grpc_error* maybe_link_callout(grpc_metadata_batch* batch,
                                       grpc_linked_mdelem* storage) {
-  grpc_metadata_batch_callouts_index idx =
-      GRPC_BATCH_INDEX_OF(GRPC_MDKEY(storage->md));
-  if (idx == GRPC_BATCH_CALLOUTS_COUNT) {
+  intptr_t idx = GRPC_STATIC_METADATA_INDEX(GRPC_MDKEY(storage->md));
+  if (idx >= GRPC_BATCH_CALLOUTS_COUNT || idx < 0) {
     return GRPC_ERROR_NONE;
   }
```

Comment:
this is ok to keep, but "==" is by-default likely. so you don't need to add that.

Reply:
What are you referring to specifically? The reason I went from == to (>= enum_max || < 0) was to guard against the negative offset case (which isn't a problem in GRPC_BATCH_INDEX_OF as originally implemented since it used some ternary ifs to constrain the range of the output.